fw/apm.c

```diff
@@ -1725,16 +1870,12 @@ tfw_cfgop_apm_server_failover(TfwCfgSpec *cs, TfwCfgEntry *ce)
 	if (tfw_cfg_check_range(tframe, 1, USHRT_MAX))
 		return -EINVAL;
 
-	hm_entry = kzalloc(sizeof(TfwApmHMCfg), GFP_KERNEL);
+	hm_entry = tfw_amp_create_hm_entry();
 	if (!hm_entry)
 		return -ENOMEM;
-
-	INIT_LIST_HEAD(&hm_entry->list);
-	hm_entry->code = code;
+	tfw_hm_entry_set_code(hm_entry, code);
```

Comment:
In my opinion, this whole dance around `tfw_hm_cfg_200_created` flag looks overblown. Just assign `hm_entry->code` here, like you do with the other members. It will looks nice and simple. When all config is processed and it's time to create (or not) the default entry, just go over the list and look for the entry with 200 code. It's clean, simple and fast. And the resulting code will be easier to read and understand.

Reply:
Below is an example of what I meant.

Here:
```C
hm_entry->code = code;
hm_entry->limit = limit;
hm_entry->tframe = tframe;
```
In `tfw_apm_create_def_health_stat_srv()`:
```C
list_for_each_entry(ent, &tfw_hm_codes_list, list) {
    if (ent->code == 200)
        break;
}
if (!list_entry_is_head(ent, &tfw_hm_codes_list, list))
    return 0;
if (!(ent = tfw_amp_create_hm_entry()))
    return -ENOMEM;
ent->code = 200;
```

Thus, no need for `tfw_hm_entry_set_code()` which obscures the initialization of the entry. The initialization is now very visible, simple, clean. Also, no need for globally defined `tfw_hm_cfg_200_created` that is only used for this specific purpose and nothing else.